code/Test_definitions/connected-network-type-subscriptions.feature

```diff
@@ -200,19 +200,40 @@ Feature: CAMARA Connected Network Type Subscriptions API, vwip - Operations on s
     And the response property "$.code" is "UNAUTHENTICATED"
     And the response property "$.message" contains a user friendly text
 
-  @connected_network_type_subscriptions_19_invalid_inconsistent_access_token
+  @connected_network_type_subscriptions_19_permission_denied
   Scenario: subscription creation with inconsistent access token for requested events subscription
-   # To test this, a token have to be obtained for a different device
+    # To test this, a token does not have the required scope
+    Given a valid subscription request body
+    And the request body property "$.device" is set to a valid testing device supported by the service
+    And header "Authorization" set to access token referring different scope
+    When the request "createDeviceReachabilityStatusSubscription" is sent
+    Then the response property "$.status" is 403
+    And the response property "$.code" is "PERMISSION_DENIED"
+    And the response property "$.message" contains a user friendly text
+
+  @connected_network_type_subscriptions_20_invalid_token_context
+  Scenario: subscription creation with invalid access token context for requested events subscription
+    # To test this, a token does not have the required device identifier
     Given a valid subscription request body
     And the request body property "$.device" is set to a valid testing device supported by the service
     And header "Authorization" set to access token referring different device
-    When the request "createSubscription" is sent
-    Then the response code is 403
-    And the response property "$.status" is 403
+    When the request "createDeviceReachabilityStatusSubscription" is sent
+    Then the response property "$.status" is 403
+    And the response property "$.code" is "INVALID_TOKEN_CONTEXT"
+    And the response property "$.message" contains a user friendly text
+
+  @connected_network_type_subscriptions_21_inconsistent_access_token_for_requested_events_subscription
+  Scenario: subscription creation with invalid access token for requested events subscription
+    # To test this, use a event type in the request which cannot be managed in this API
+    Given a valid subscription request body
+    And the request body property "$.device" is set to a valid testing device supported by the service
+    And the request body property "$.types" contains an unsupported event type in this API
```

Comment:
This is not exactly that.
This is when you trigger a subscription request for event type A with a token associated with event type B

Let give you an example:

Suppose in the `/bc-authorize` the` purpose= openid dpv:xxxxxx device-roaming-status-subscriptions:org.camaraproject.roaming-status-subscriptions.v0.roaming-status:create`

and then you making the `POST /subscription` with `even-type = org.camaraproject.connected-network-type-subscriptions.v0.network-type-changed`

Reply:
Thanks again for the review.
Oh yes, I agree - otherwise it would be the same as test case no. 11.
Adjustments are made in all three subscription APIs.